sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -677,14 +696,27 @@ case class StructsToJson(
   override def checkInputDataTypes(): TypeCheckResult = child.dataType match {
     case _: StructType | ArrayType(_: StructType, _) =>
       try {
-        JacksonUtils.verifySchema(rowSchema)
+        JacksonUtils.verifySchema(rowSchema.asInstanceOf[StructType])
+        TypeCheckResult.TypeCheckSuccess
+      } catch {
+        case e: UnsupportedOperationException =>
+          TypeCheckResult.TypeCheckFailure(e.getMessage)
+      }
+    case ArrayType(_: MapType, _) =>
+      TypeCheckResult.TypeCheckSuccess
+    case MapType(_: DataType, st: StructType, _: Boolean) =>
+      try {
+        JacksonUtils.verifySchema(st)
         TypeCheckResult.TypeCheckSuccess
       } catch {
         case e: UnsupportedOperationException =>
           TypeCheckResult.TypeCheckFailure(e.getMessage)
       }
+    case _: MapType =>
```

Comment:
Not all map types are valid for JSON. We still need verify the type.

Reply:
We need to revise `JacksonUtils` to check map type too.

Reply:
@viirya But the `JacksonUtils.verifySchema` only verify a `StructType`, so we need to add a new one?

Reply:
Maybe we can just do something like:

    case mt: MapType =>
      val st = StructType(StructField("a", mt) :: Nil)
      JacksonUtils.verifySchema(st)

Reply:
oh yes. It's a tricky way but make sense. =D

Reply:
Yeah, it is. Maybe we should revise `JacksonUtils` to be able to verify `MapType` directly. But I don't see `JacksonUtils.verifySchema` used in other place. We can leave it as TODO if you want to.

Reply:
Basically I don't like to increase the size of this PR anymore.

Reply:
ok. I just leave a TODO comment for it.